src/app/services/image/image.service.ts

```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export abstract class ImageService {
+  abstract loadImage(path: string): string;
+
+  abstract uploadImage(image: string): Observable<string>;
```

Comment:
Please add documentation comments to these methods since they will be reused across multiple files. You only have to do it in the abstract service class, not the concrete service classes.
```ts
/** I am a documentation comment. Notice I start with double stars */
```